core/trino-main/src/test/java/io/trino/sql/TestExpressionInterpreter.java

```diff
@@ -1430,6 +1433,38 @@ public void testLiterals()
         assertEquals(optimize("X'1234'"), Slices.wrappedBuffer((byte) 0x12, (byte) 0x34));
     }
 
+    @Test
+    public void testEvaluateFailingExpression()
+    {
+        assertTrinoExceptionThrownBy(() -> evaluate("0/0"))
+                .hasErrorCode(DIVISION_BY_ZERO);
+
+        assertTrinoExceptionThrownBy(() -> evaluate("fail('Some message')"))
+                .hasErrorCode(GENERIC_USER_ERROR)
+                .hasMessage("Some message");
+
+        assertTrinoExceptionThrownBy(() -> evaluate("IF(0 > 1, 0/1, 0/0)"))
+                .hasErrorCode(DIVISION_BY_ZERO);
+
+        assertTrinoExceptionThrownBy(() -> evaluate("IF(0/0 > 0, true, false)"))
+                .hasErrorCode(DIVISION_BY_ZERO);
+
+        assertTrinoExceptionThrownBy(() -> evaluate("coalesce(2/2, 1/1, 0/0)"))
```

Comment:
This shouldn't fail. Per the SQL spec, it's syntactic sugar for:

```
CASE 
  WHEN 2/2 IS NOT NULL THEN 2/2
  WHEN 1/1 IS NOT NULL THEN 1/1
  ELSE 0/0
END
```

The evaluation of the bodies is "lazy", so `0/0` should never be evaluated.